src/com/owncloud/android/MainApp.java

```diff
@@ -117,4 +120,24 @@ public static String getLogName() {
         return getAppContext().getResources().getString(R.string.log_name);
     }
 
+    // user agent
+    public static String getUserAgent() {
+        String appString = getAppContext().getResources().getString(R.string.user_agent);
+        String packageName = getAppContext().getPackageName();
+        PackageInfo pInfo = null;
+        try {
+            pInfo = getAppContext().getPackageManager().getPackageInfo(packageName, 0);
+        } catch (PackageManager.NameNotFoundException e) {
+
+        }
+        String version = "";
+        if (pInfo != null) {
+            version = "/" + pInfo.versionName;
```

Comment:
This should go inside the previous try ... catch block to avoid a NullPointerException in ```pInfo.versionname```